src/Swarm/Game/Recipe.hs

```diff
@@ -117,11 +119,11 @@ resolveRecipes em = (traverse . traverse) (\t -> maybe (Failure [t]) Success (lo
 
 -- | Given an already loaded 'EntityMap', try to load a list of
 --   recipes from the data file @recipes.yaml@.
-loadRecipes :: MonadIO m => EntityMap -> m (Either Text [Recipe Entity])
-loadRecipes em = runExceptT $ do
-  fileName <- liftIO $ getDataFileName "recipes.yaml"
-  res <- liftIO $ decodeFileEither @[Recipe Text] fileName
-  textRecipes <- res `isRightOr` (from . prettyPrintParseException)
+loadRecipes :: (Has (Lift IO) sig m) => EntityMap -> m (Either Text [Recipe Entity])
```

Comment:
Eventually I'd love to rewrite things like this that use `IO` in terms of some finer-grained custom effects.  E.g. in this case we could have a custom effect for loading data files.  But we don't have to do it in this PR.  In fact it might be nicer to just get something simple that works and then worry about fancy details later.

Reply:
Agreed, my goal for this PR was to convert the code to fused-effects with the smallest code changes possible. We can then have one PR per new effect that we introduce. It sure would be great if logging could be an effect for instance.